src/features/phone_applications_foreground/rapids/main.py

```diff
@@ -113,21 +125,19 @@ def process_app_features(data, requested_features, time_segment, provider, filte
 
 def rapids_features(sensor_data_files, time_segment, provider, filter_data_by_segment, *args, **kwargs):
     
-    apps_events_data = pd.read_csv(sensor_data_files["sensor_data"])
-    requested_events_features = provider["FEATURES"]["APP_EVENTS"]
-    
     app_episodes_requirement = provider["INCLUDE_EPISODE_FEATURES"]
```

Comment:
Hi @jenniferfedor, I think it would be more clear if we assign `episodes_data` or `event_data` to the `sensor_data` variable of [features.smk](https://github.com/carissalow/rapids/blob/feature/fix_phone_app_foreground_bugs/rules/features.smk#L127) based on `provider["INCLUDE_EPISODE_FEATURES"]` directly. It could be implemented by creating a function in [common.smk](https://github.com/carissalow/rapids/blob/feature/fix_phone_app_foreground_bugs/rules/common.smk) to get the input. Then we can call the process_app_features() function only once. What do you think, @jenniferfedor and @JulioV ?

Reply:
The function to get the input file could be similar to [get_locations_python_input()](https://github.com/carissalow/rapids/blob/master/rules/common.smk#L24).

Reply:
Hi @Meng6 and @JulioV, thank you so much for your suggestions! I’ve updated the following:  
- Add a `get_applications_foreground_input()` function to `common.smk`  
- Update `sensor_data` and remove `episode_data` inputs in `features.smk`  
- Refactor `main.py` so we call `process_app_features()` only once 
- Give the timestamp column variable a more descriptive name   

Could you please review whenever you have some time and let me know if there is anything else you’d like me to change? One of our interns is working on updating our tests. Thanks!